### DISCUSSION TITLE
Future SNES CIC suggestion

### DISCUSSION THREAD

**Discussion by Mar-uri** [sanni/cartreader#446] [Ideas & Suggestions]
Hi, sanni - san
( “- san” means “Mr. / Ms” in Japanese)
 
First of all, I appreciate you to open OSCR design on Github.  I have already made and enjoy Ver.3 and Ver.5 OSCR.

Then I have a request to sanni-san for designing future SNES CIC for OSCR Ver.5.

I suggest to have both pattern for PIC12F629 on “chip”(SOIC8 variant) and “DIP IC” (DIP 8 variant) due to lack of production capacity of semiconductor.

![F87DF26A-A4B2-4007-A950-141F25274706](https://user-images.githubusercontent.com/110158697/181743274-2c93e9b6-a746-4df9-b9a4-9ac9ce6fcfc2.jpeg)

![85089C0E-27AC-4A69-8AD7-089F52A5608E](https://user-images.githubusercontent.com/110158697/181743268-d858c2fd-2b9e-4f3b-8b2b-549b6a20b88b.jpeg)

——————-

And in addition, if possible, OSCR can have capability of program (reprogram) for PIC12F629. It would be very helpful for OSCR users, I believe.
But, I do not know technical design for PIC, sorry.

**Comment by Ancyker** [comment]
Doesn't it already have that (DIP/SOIC)? -> https://prnt.sc/PstoCxIoNFBC

Also, it's in stock at the place linked on the wiki, https://www.buyicnow.com/it.php?i=114

**Reply by Mar-uri** [reply]
I have DIP8 and do not have SOIC.

In Japan, Tokyo area has Akihabara town. Almost all parts buy at Alihabara.  But, recently SOIC PIC are out of stock in some shops.  

Beside, many PIC writers are supported DIP type PIC, basically.  SOIC write is one of option which means cost requirement.

**Reply by Mar-uri** [reply]
Sorry, I misinderstood your comment.

**Reply by Ancyker** [reply]
> Sorry, I misinderstood your comment.

It's ok. 😊

**Comment by sanni** [comment]
I updated the Wiki to resemble the current state: https://github.com/sanni/cartreader/wiki/Flashing-the-snesCIC

I also added a link to a DIY Arduino PIC programmer. 

Integrating the function of flashing PICs into the OSCR would be out of the scope of a Cartridge Dumper but you can use the Arduino that you bought for it to flash the PIC if you build that DIY PIC programmer. Not sure if it's worth the effort though since you can just buy the PIC preflashed.

**Reply by Mar-uri** [reply]
Thank you for your reply. 
I agree your scope for OSCR due to out of memory, maybe.   And very good anothet suggestion is  happy for me(us).

**Reply by Ancyker** [reply]
@sanni is there any specific testing that could be done with the CIC/PIC that you need?

**Comment by Mar-uri** [comment]
This is SOIC and SOP both pattern image.
If it wii be easy to add, I hope future SNES CIC having both pattern.

![BB6CE400-541C-467F-8359-57F1F351AB5E](https://user-images.githubusercontent.com/110158697/182020333-536912fd-f711-4d96-9113-381cab27a74e.jpeg)

**Reply by Mar-uri** [reply]
Sorry, I wrote this comment before looking SNES CIC.  What’s a great idea!  I appreciate your actions.
